tools/bin/mbedtls-rewrite-branch-style

```diff
@@ -477,7 +496,23 @@ class CodeStyleBranchRewriter(MbedTLSBranchRewriter):
         # cherry-picked commit, but might mess up modified files. Then
         # we override the content of modified files, and finally we restyle
         # files that need restyling.
-        self.run_git(['cherry-pick', '-Xtheirs', '--allow-empty', new_commit])
+        if deleted_styled_files:
+            # There are some deleted files, so expect cherry-pick to return
+            # with exit status 1.
+            try:
+                self.run_git(['cherry-pick', '-Xtheirs', '--allow-empty', new_commit])
+            except subprocess.CalledProcessError as c:
```

Comment:
I don't like catching an error this way, because it could also catch other errors. Can we make `cherry-pick` not error out if there are deleted files?

Reply:
[Passing `--keep-redundant-commits` to `cherry-pick`](https://github.com/Mbed-TLS/mbedtls-docs/pull/71#discussion_r1066118119) is enough to make `test_6882` pass, but not `teest_rename_delete`. Given that the real-world case passes, I'm inclined to consider that good enough if we can't find a better solution quickly.

Reply:
Ah I should perhaps have documented more clearly - `test_6882` is just another testcase I added to increase the total number, it's unrelated to deleted files.

Reply:
To make `cherry-pick` not return an error would require some kind of `--return-success-on-conflicts` command-line switch, which I've looked for and not found. What I have done is rethrown the error if the return value is anything other than 1, which will catch some if not all other errors.

Reply:
Ah, my bad, I should have looked.

Maybe the right thing would be to assemble the commit object manually... That's a bit too low-level for my tastes though.

Alternatively, perhaps try `ours` as the merge strategy, then figure out what to do for added and deleted files (including renames/copies)?

The right solution is actually probably to treat this as an interactive rebase and give the user the opportunity to resolve the conflict manually. That requires the script to carry on with its original state though. I'd actually have liked to make the process `git rebase --exec=/path/to/our/script`, but I couldn't figure out how to arrange what I want this way.

Reply:
Hmm
I suppose thanks to the `reset`-ing later on, the actual state of the tree after the `cherry-pick` does not matter, we're just doing it to retain commit details. In that case I suppose `-Xours` might fix the whole issue (since it cannot fail, as far as I'm aware).

Reply:
According to my notes, `-Xtheirs`

> ensures that added and removed files are according to the cherry-picked commit, but might mess up modified files

but I'm not fully confident. In fact, the fact that my code didn't work with deleted files shows that it isn't quite doing what I think it does.

Reply:
I've been experimenting a little with `--strategy=ours` (I think you meant this — `-Xours` can fail, and in fact does fail just like `-Xtheirs` in the “Stop testing Aria” commit). That does avoid conflicts, but I don't see an easy way to handle added/renamed files (and likely mode changes would disappear, too).

Reply:
Ah yes I think that is what I meant. I changed my test copy to that but as you say, it fell over on a rename. It might be fixable by reordering the `git reset`s somehow, but I'm not quite sure how.

Alternatively, we could take the loss of some error reporting as an acceptable tradeoff given the complication of other approaches and stick with what is in this PR.

Reply:
What bothers me is that we're losing error _detection_, not just error _reporting_. As in, we're potentially quietly constructing wrong output, and not just erroring with an incomprehensible message.

I'm looking for a way to say “if there's a deleted file in their branch, just delete it” but no luck so far.

Reply:
Hmmm, maybe delete the file _before_ the cherry-pick? And then squash the cherry-pick commit, but arranging to pick the cherry-pick commit's non-content data rather than the delete commit.

Reply:
> maybe delete the file before the cherry-pick?

Done in https://github.com/Mbed-TLS/mbedtls-docs/pull/74